api/views.py

```diff
@@ -91,12 +91,31 @@ def phone(self, request, format=None):
         outserializer = UserAccessSerializer(user)
         return Response(outserializer.data)
 
-    #    @action(detail=False, method=["post"])
+    @action(detail=False, methods=["post"], throttle_classes=[VerySlowThrottle])
     def nfc(self, request, format=None):
         """
-        TODO: implement nfc reader serializer
+        NFC card access
         """
-        return Response(status=status.HTTP_501_NOT_IMPLEMENTED)
+        inserializer = AccessDataSerializer(data=request.data)
+        inserializer.is_valid(raise_exception=True)
+
+        cardid = inserializer.validated_data.get("payload")
+        qs = NFCCard.objects.filter(cardid=cardid)
+
+        if qs.count() == 0:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        # our user
+        user = qs.first().user
+        # user does not have access rights
+        if not user.is_active:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        if not user.has_door_access():
```

Comment:
Why not check here if card.subscription.state is active? That would be more generic.